js/src/logger.ts

```diff
@@ -1005,6 +1063,26 @@ export class Logger<IsAsyncFlush extends boolean> implements Exportable {
     logFeedbackImpl(this.state, this.parentObjectType(), this.lazyId, event);
   }
 
+  /**
+   * Update a span in the experiment using its id. It is important that you only update a span once the original span has been fully written and flushed,
+   * since otherwise updates to the span may conflict with the original span.
+   *
+   * @param event The event data to update the span with. Must include `id`. See `Experiment.log` for a full list of valid fields.
```

Comment:
We can enforce this in the type signature too? `Omit<Partial<ExperimentEvent>, "id"> & Required<Pick<ExperimentEvent>, "id">` or something?